integration-tests/src/test_loop/builder.rs

```diff
@@ -28,20 +28,32 @@ use near_network::test_loop::{TestLoopNetworkSharedState, TestLoopPeerManagerAct
 use near_parameters::RuntimeConfigStore;
 use near_primitives::epoch_manager::EpochConfigStore;
 use near_primitives::network::PeerId;
+use near_primitives::sharding::ShardChunkHeader;
 use near_primitives::test_utils::create_test_signer;
 use near_primitives::types::{new_shard_id_tmp, AccountId};
 use near_store::adapter::StoreAdapter;
 use near_store::config::StateSnapshotType;
 use near_store::genesis::initialize_genesis_state;
 use near_store::test_utils::{create_test_split_store, create_test_store};
-use near_store::{Store, StoreConfig, TrieConfig};
+use near_store::{ShardUId, Store, StoreConfig, TrieConfig};
+use near_vm_runner::logic::ProtocolVersion;
 use near_vm_runner::{ContractRuntimeCache, FilesystemContractRuntimeCache};
 use nearcore::state_sync::StateSyncDumper;
 use tempfile::TempDir;
 
 use super::env::{ClientToShardsManagerSender, TestData, TestLoopChunksStorage, TestLoopEnv};
 use super::utils::network::{chunk_endorsement_dropper, partial_encoded_chunks_dropper};
 
+enum DropCondition {
+    /// Whether test loop should drop all chunks validated by the given account.
+    ChunksValidatedBy(AccountId),
+    /// Whether test loop should drop all endorsements from the given account.
+    EndorsementsFrom(AccountId),
+    /// Whether test loop should drop all chunks in the given range of heights
+    /// relative to first block height where protocol version changes.
+    ProtocolUpgradeChunkRange((ProtocolVersion, HashMap<ShardUId, std::ops::Range<i64>>)),
```

Comment:
Is this meant to happen on every protocol upgrade or just the first one? Assuming it is desirable to have multiple protocol upgrades in one test

Reply:
Should work for many upgrades, but we never had such tests and I don't think they would be very useful.

Reply:
Right. For now I have see many issues on resharding side...